clients/pkg/promtail/wal/writer.go

```diff
@@ -27,6 +27,14 @@ const (
 	minimumCleanSegmentsEvery = time.Second
 )
 
+// WriterEventSubscriber is an interface that objects that want to receive events from the wal Writer can implement. After
+// they can subscribe to events by adding themselves as subscribers on the Writer with writer.Subscribe.
+type WriterEventSubscriber interface {
+	// NotifySegmentsCleaned will notify the subscriber that all WAL segments with a number lower or equal to num, were
+	// reclaimed.
+	NotifySegmentsCleaned(num int)
+}
+
```

Comment:
couldn't this just be the `WriteTo` interface we already have? we could just call `subscribers.SeriesReset`